shared/src/main/scala-2.13/zipper/ForImpl.scala

```diff
@@ -0,0 +1,32 @@
+package zipper
+
+import shapeless.{Generic, HList}
+import shapeless.ops.hlist.{Replacer, Selector}
+
+import scala.collection.Factory
+
+private[zipper] trait ForImpl {
```

Comment:
Unfortunately, I had to copy-paste the whole `For` class for Scala 2.12 and 2.13 since `CanBuildFrom` was cut down in Scala 2.13. It's a little bit messy but super straightforward.